packages/server/src/server.ts

```diff
@@ -137,9 +142,18 @@ export const createApp = async (options: CreateAppOptions): Promise<{
 
 		if (!fs.existsSync(selfSignedKeyPath) || !fs.existsSync(selfSignedCertPath)) {
 			try {
+				let altNames: string[] = [];
+				let networkInterfaces = os.networkInterfaces();
+				for(let ifName of Object.keys(networkInterfaces)){
+					let interfaces = networkInterfaces[ifName];
+					interfaces.filter(i => !i.internal && i.family === "IPv4").forEach(i => {
+						altNames.push(i.address);
+					});
+				}
 				const certs = await new Promise<pem.CertificateCreationResult>((res, rej): void => {
 					pem.createCertificate({
 						selfSigned: true,
+            altNames: altNames
```

Comment:
I don't think this should be automatic. Since the certificate is self signed, your computer won't trust alt names anyway.

Reply:
If you decide to trust the self-sigend certificate, the hostname/ip of the certificate still has to match in order to pass for a secure TLS connection.

Reply:
Are you positive? We've been using this for a while now without any alt names and no issues have been reported.

Reply:
I'm running this behind a proxy (on a different server) which reëncrypts the connection to code-server. The proxy trusts the self-signed certificate of code-server. But it is failing on the hostname verification, because the external ip of code-server is not matching the "localhost" which is in the certificate.

Reply:
That's dangerous imo as you can get MITMed easily with a self signed certificate. It's best you generate the certificate yourself and add it as a trusted certificate to the proxy or use HTTP and move code-server onto the proxy server.

Reply:
I guess my same reasoning applies in general to code-server generating self signed certificates. 🤔